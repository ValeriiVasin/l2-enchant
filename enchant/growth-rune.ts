import { createEnchantMap } from '../helpers/create-enchant-map';
import { itemsRange } from '../helpers/items-range';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const growthRuneEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.Drop,
  enchantItem: Item.GrowthRuneFragment,
  chances: [70, 60, 50, 40, 30, 50, 40, 30, 20, 20, 30, 20, 15, 10],
  items: itemsRange(Item.GrowthRune_1, Item.GrowthRune_15),
  boundaries: new Set([Item.GrowthRune_6, Item.GrowthRune_11]),
});
