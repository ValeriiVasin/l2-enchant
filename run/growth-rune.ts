import { itemsRange } from '../helpers/items-range';
import { run } from '../helpers/run';
import { Item } from '../items';

run(Item.GrowthRune_1, itemsRange(Item.GrowthRune_2, Item.GrowthRune_15), 100_000);
