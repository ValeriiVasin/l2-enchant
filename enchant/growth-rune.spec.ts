import { growthRuneEnchantMap } from './growth-rune';

it('enchant map: growth rune', () => {
  expect(growthRuneEnchantMap).toMatchSnapshot();
});
