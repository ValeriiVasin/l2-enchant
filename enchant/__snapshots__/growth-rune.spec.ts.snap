// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: growth rune 1`] = `
Map {
  "Руна Развития Ур.1" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.1",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Руна Развития Ур.2",
      },
      Object {
        "chance": 30,
        "item": "Руна Развития Ур.1",
      },
    ],
  },
  "Руна Развития Ур.2" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.2",
    "results": Array [
      Object {
        "chance": 60,
        "item": "Руна Развития Ур.3",
      },
      Object {
        "chance": 40,
        "item": "Руна Развития Ур.1",
      },
    ],
  },
  "Руна Развития Ур.3" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.3",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Руна Развития Ур.4",
      },
      Object {
        "chance": 50,
        "item": "Руна Развития Ур.1",
      },
    ],
  },
  "Руна Развития Ур.4" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.4",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Руна Развития Ур.5",
      },
      Object {
        "chance": 60,
        "item": "Руна Развития Ур.1",
      },
    ],
  },
  "Руна Развития Ур.5" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.5",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Руна Развития Ур.6",
      },
      Object {
        "chance": 70,
        "item": "Руна Развития Ур.1",
      },
    ],
  },
  "Руна Развития Ур.6" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.6",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Руна Развития Ур.7",
      },
      Object {
        "chance": 50,
        "item": "Руна Развития Ур.6",
      },
    ],
  },
  "Руна Развития Ур.7" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.7",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Руна Развития Ур.8",
      },
      Object {
        "chance": 60,
        "item": "Руна Развития Ур.6",
      },
    ],
  },
  "Руна Развития Ур.8" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.8",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Руна Развития Ур.9",
      },
      Object {
        "chance": 70,
        "item": "Руна Развития Ур.6",
      },
    ],
  },
  "Руна Развития Ур.9" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.9",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Руна Развития Ур.10",
      },
      Object {
        "chance": 80,
        "item": "Руна Развития Ур.6",
      },
    ],
  },
  "Руна Развития Ур.10" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.10",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Руна Развития Ур.11",
      },
      Object {
        "chance": 80,
        "item": "Руна Развития Ур.6",
      },
    ],
  },
  "Руна Развития Ур.11" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.11",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Руна Развития Ур.12",
      },
      Object {
        "chance": 70,
        "item": "Руна Развития Ур.11",
      },
    ],
  },
  "Руна Развития Ур.12" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.12",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Руна Развития Ур.13",
      },
      Object {
        "chance": 80,
        "item": "Руна Развития Ур.11",
      },
    ],
  },
  "Руна Развития Ур.13" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.13",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Руна Развития Ур.14",
      },
      Object {
        "chance": 85,
        "item": "Руна Развития Ур.11",
      },
    ],
  },
  "Руна Развития Ур.14" => Object {
    "cost": "Фрагмент Руны Развития",
    "item": "Руна Развития Ур.14",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Руна Развития Ур.15",
      },
      Object {
        "chance": 90,
        "item": "Руна Развития Ур.11",
      },
    ],
  },
}
`;
